### ISSUE TITLE
Add known metadata keys to autosuggestion in MetadataPane

### ISSUE THREAD

**Issue by jmduke** [buttondown-email/roadmap#950]
## TL;DR

Make it easier for folks to manually add metadata to emails and subscribers in the UI by suggesting autocompleted keys in the metadata input based on existing metadata they've set.

## Rough plan of attack

- [ ] Create an action on the newsletter model, `generate_metadata_keys`, that takes a `newsletter_id` and a `model_type` (`"email" | "subscriber"`) and returns a list of keys. 
- [ ] Invoke that action and serialize it onto the main `newsletter` JSON payload that is accessible to the frontend
- [ ] Read the serialized key caches in Vue and use them to provide autocomplete for:
  - [ ] emails
  - [ ] subscribers

## The path not taken

As I began to spec this out, I thought the best approach would be _denorming_ the keys rather than generating them inline, creating a model like:

```
class MetadataKeyCache(BaseModel):
  class ModelType(Enum):
    SUBSCRIBERS = "subscribers"
    EMAILS = "emails"
  newsletter_id: uuid
  model_type: ModelType
  keys: list[str]
```

and keeping that updated asynchronously through signals. It turns out this is unnecessary; I thought a query would be prohibitively slow, but running something like:

```
select distinct jsonb_object_keys(metadata)
from emails_subscriber
where newsletter_id = '80e359a9-f0af-4c5d-a990-296aac306e99'
```

on even Buttondown's larger 100K+ subscriber accounts is fairly performant (~150ms).

## Other notes

- While we don't need to worry about it in the interface, these metadata keys _will_ be used elsewhere: in particular, I want to expand list views to allow authors to add columns for arbitrary metadata (https://github.com/buttondown-email/roadmap/issues/340) and this model can be used there
- There are probably some interesting questions around weird edge cases (should we try and _delete_ keys from the cache if metadata is removed?) I think we can punt on those for now; it feels very edge-case-of-an-edge-case-y.
- There are two performance questions that are not immediately obvious to me. I outlined the _easiest_ path that sacrifices a bit of performance for ease of use; I'm not 100% sold, but neither choice is a huge lift.
  - Should we cache the metadata keys onto the Newsletter model, refreshing them after a certain point in time? 
  - Should we actually _serialize_ the key list onto the newsletter payload, or should we fetch it asynchronously upon load?
